runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -135,11 +138,13 @@ public SparkPipelineResult run(Pipeline pipeline) {
             isOneOf(PipelineResult.State.STOPPED, PipelineResult.State.DONE));
 
         // validate assertion succeeded (at least once).
-        int successAssertions = 0;
-        try {
-          successAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
-        } catch (NullPointerException e) {
-          // No assertions registered will cause an NPE here.
+        long successAssertions = 0;
+        Iterable<MetricResult<Long>> counterResults = result.metrics().queryMetrics(
+            MetricsFilter.builder()
+                .addNameFilter(MetricNameFilter.named(PAssert.class, PAssert.SUCCESS_COUNTER))
+                .build()).counters();
+        for (MetricResult<Long> counter : counterResults) {
+          successAssertions++;
```

Comment:
This is missing the conditional mentioned by @bjchambers in his original review. i.e.
```
if (counter.attempted().longValue() > 0) {
  successAssertions++;
}
```